libcontainer/capabilities/capabilities.go

```diff
@@ -24,47 +24,69 @@ func init() {
 	}
 }
 
-// New creates a new Caps from the given Capabilities config.
+// New creates a new Caps from the given Capabilities config. Unknown Capabilities
+// or Capabilities that are unavailable in the current environment are ignored,
+// printing a warning instead.
 func New(capConfig *configs.Capabilities) (*Caps, error) {
 	var (
-		err  error
-		caps Caps
+		err      error
+		caps     Caps
+		ignored  []string
+		warnings = make([]string, 0)
 	)
 
-	if caps.bounding, err = capSlice(capConfig.Bounding); err != nil {
-		return nil, err
+	if caps.bounding, ignored = capSlice(capConfig.Bounding); len(ignored) > 0 {
+		warnings = append(warnings, ignored...)
```

Comment:
IMO the best way to de-dup the list of unknown caps is to put them into a `map[string]struct{}{}` right away.

Say `capSlice` may have a second argument with the `ignored` map, to which it would add more entries.

```go
// ...
// Names of unknown or unavailable capabilities are added to the ignored map.
func capSlice(caps []string, ignored map[string]struct{}{}) []capability.Cap {
```